src/Multicall3.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.12;
+
+/// @title Multicall3
+/// @notice Aggregate results from multiple read-only function calls
+/// @dev Multicall & Multicall2 backwards-compatible
+/// @author Michael Elliot <mike@makerdao.com>
+/// @author Joshua Levine <joshua@makerdao.com>
+/// @author Nick Johnson <arachnid@notdot.net>
+/// @author Andreas Bigger <andreas@nascent.xyz>
+
+contract Multicall3 {
+    struct Call {
+        address target;
+        bytes callData;
+    }
+
+    struct Call3 {
+        address target;
+        bytes callData;
+        bool requireSuccess;
+    }
+
+    struct Result {
+        bool success;
+        bytes returnData;
+    }
+
+    function aggregate(Call[] memory calls) public returns (uint256 blockNumber, bytes[] memory returnData) {
+        blockNumber = block.number;
+        uint256 length = calls.length;
+        returnData = new bytes[](length);
+        for (uint256 i = 0; i < length;) {
```

Comment:
There is no need (except for readability) to initialise the default value 0 for the incrementer `i`. Otherwise, you need an additional `MSTORE` operation. Recent Solidity compilers may be able to optimize this away but older ones not afaik.

Reply:
Doesn’t hurt and fairly standard practice, I’ve tested and the compiler does optimize

Reply:
Cool, yea @abigger87 also thought the compiler would optimize, so since you both tested let's leave this as-is for clarity

Reply:
did a quick test to confirm initializing to zero does not impact gas cost